x-pack/plugins/cases/server/services/user_profiles/index.ts

```diff
@@ -41,6 +48,101 @@ export class UserProfileService {
     this.options = options;
   }
 
+  public async findAssignees(
+    request: KibanaRequest,
+    casesClient: CasesClient
+  ): Promise<UserProfile[]> {
+    const params = pipe(
+      excess(FindAssigneesRequestRt).decode(request.query),
+      fold(throwErrors(Boom.badRequest), identity)
+    );
+
+    const { searchTerm, size, owners } = params;
+
+    try {
+      if (this.options === undefined) {
+        throw new Error('UserProfileService must be initialized before calling suggest');
+      }
+
+      const { spaces } = this.options;
+
+      const securityPluginFields = {
+        securityPluginSetup: this.options.securityPluginSetup,
+        securityPluginStart: this.options.securityPluginStart,
+      };
+
+      UserProfileService.validateSizeParam(size);
+
+      if (!UserProfileService.isSecurityEnabled(securityPluginFields) || owners.length <= 0) {
+        return [];
+      }
+
+      const assigneesWithinAllCases = new Set(
+        await casesClient.cases.getAssignees({ owner: owners })
+      );
+
+      const { securityPluginStart } = securityPluginFields;
+
+      // if we didn't receive a search term just return all the assignees within cases
+      if (isEmpty(searchTerm)) {
+        const assignedUserProfiles = await securityPluginStart.userProfiles.bulkGet({
+          uids: assigneesWithinAllCases,
+          dataPath: 'avatar',
+        });
+
+        return assignedUserProfiles.slice(0, size ?? MAX_PROFILES_SIZE);
+      }
+
+      const suggestedUsers = await UserProfileService.suggestUsers({
```

Comment:
I understand the reasoning behind the technical decisions but maybe we overcomplicate things. I am thinking that maybe is better to return all assignees (turn the find to a getAllAssignees as we do for the tags) and do the search on the front end. The [`EuiSelectable`](https://elastic.github.io/eui/#/forms/selectable) supports searching over a list of options. I [tested](https://codesandbox.io/s/quizzical-haze-k4cv49?file=/demo.tsx) it with 10K elements and it seems pretty performant (my concern is if searching over 10K elements will be slow on the browser). What do you think? We can talk it offline if you want.